spline-shadow/persistence/src/main/scala/za/co/absa/spline/persistence/model/persistenceModel.scala

```diff
@@ -17,15 +16,75 @@
 
 package za.co.absa.spline.persistence.model
 
-case class Progress(timestamp: Long, readCount: Long, _key: Option[String] = None, _id: Option[String] = None, _rev: Option[String] = None) extends DocumentOption
+import DataType._
+
+case class Progress(timestamp: Long, readCount: Long,
+  key: Option[String] = None, _id: Option[String] = None, _rev: Option[String] = None) extends DocumentOption
```

Comment:
_type is a type hint,

> Not sure I understand the purpose of designing `DataType` hierarchy that way.
> In your hierarchy you use the same three concrete data type subclasses as in the original model (`SimpleType`, `ArrayType` and `StructType`), but their properties are now mixed up.
> 
>     * What is the benefit of making `name` to be a common property of all three types, while it only has a meaning  for `SimpleType` and for the other two it simply duplicates the `_type` property?
> 
>     * Why does `ArrayType` being a homogenous data structure contains a sequence of children types? What are those children?
> 
>     * Why to decompose a sequence of `StructType`'s fields (where a field has a name and type) into two highly coupled sequences, one with only names and the other with only respected types, having them relying on the sort order to maintain the dependency between the two?

- _type is a type hint for deserialization frameworks. It comes from PolymorphicDocumentOption. 
- name is a field I would expect to be used on the frontend or in queries. Since simple types dont contain "Struct" and "Array" there wont't be any clash.
- ArrayType has an array such that it has similar interface to StructType akin to original class which has a method Array#childDataTypeIds. This symmetry should make implementation of AQL queries and frontend simpler.

Reply:
How will it make it simpler? Can you give an example of AQL query or Angular service where such design will be helpful?

Reply:
Intent was to have united schema e.g. there is method in our original code base ```spline.model.dt.Array#childDataTypeIds``` which returns an array on single value. So you dont have to write if-statements when looking for childDatatypeIds.

But the other option has also its advantages as well. So as agreed I went with the alternative approach.